src/BuildScriptGenerator/DefaultDockerfileGenerator.cs

```diff
@@ -255,5 +261,49 @@ private string ConvertToRuntimeVersion(string platformName, string platformVersi
         {
             return this.platformDetector.GetCompatiblePlatforms(ctx);
         }
+
+        private void ValidateDockerfileImages(DockerfileProperties dockerfileProperties)
+        {
+            if (string.IsNullOrEmpty(dockerfileProperties.BuildImageName))
+            {
+                this.LogInvalidDockerfileImageException(
+                    "Provided build image name parsed from the provided --build-image argument is empty. " +
+                    "Please provide a valid value for --build-image, or remove the argument to use the default " +
+                    "'cli:stable' build image from mcr.microsoft.com/oryx.");
+            }
+
+            if (string.IsNullOrEmpty(dockerfileProperties.BuildImageTag))
+            {
+                this.LogInvalidDockerfileImageException(
+                    "Provided build image tag parsed from the provided --build-image argument is empty. " +
+                    "Please provide a valid value for --build-image, or remove the argument to use the default " +
+                    "'cli:stable' build image from mcr.microsoft.com/oryx.");
+            }
+
+            if (string.IsNullOrEmpty(dockerfileProperties.RuntimeImageName))
+            {
+                this.LogInvalidDockerfileImageException(
+                    "Either the value provided to the --runtime-platform argument is empty, or the platform " +
+                    "discovered by Oryx does not have any available or supported runtime images. Please view the following " +
+                    "document for more information on runtimes supported by Oryx: " +
+                    "https://github.com/microsoft/Oryx/blob/main/doc/supportedRuntimeVersions.md");
+            }
+
+            if (string.IsNullOrEmpty(dockerfileProperties.RuntimeImageTag))
+            {
+                this.LogInvalidDockerfileImageException(
+                    "Either the value provided to the --runtime-platform-version argument is empty, or the version " +
+                    "discovered by Oryx is not available or supported for the given platform. Please view the following " +
+                    "document for more information on runtime versions supported by Oryx: " +
+                    "https://github.com/microsoft/Oryx/blob/main/doc/supportedRuntimeVersions.md");
+            }
+        }
+
+        private void LogInvalidDockerfileImageException(string message)
```

Comment:
Please consider renaming to `LogAndThrowInvalidDockerfileImageException` or something: I thought this was just logging and that the flow would continue.